src/iop/colorzones.c

```diff
@@ -806,7 +806,8 @@ static void _draw_color_picker(dt_iop_module_t *self, cairo_t *cr, dt_iop_colorz
                                const float *const picker_color, const float *const picker_min,
                                const float *const picker_max)
 {
-  if(self->request_color_pick == DT_REQUEST_COLORPICK_MODULE)
```

Comment:
I'm not sure we should remove this line. It really means that the color picker has been activated and that the color pick values are ready. It may be ok in all case and if the widget is active then the request_color_pick is set to DT_REQUEST_COLORPICK_MODULE, but I won't bet on that. Last but not least it makes it clear in the code that we are handling the color picker request here.

Reply:
Makes sense. All color picker access should really go through access functions, but the remaining places where the proxy infrastructure isn't used are sufficiently complicated that I didn't dare touch them.

My initial sub-goal for colorpicker was to move the data to one global picker rather than one per module, since only one can be active at the same time anyway. But that was just _too_ complex.

I put back the same test in tonecurve as well; assuming that's what you meant.